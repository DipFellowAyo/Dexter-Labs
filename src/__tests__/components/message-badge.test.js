import React from 'react'
import Badge from '@material-ui/core/Badge'
import Box from '@material-ui/core/Box'
import IconButton from '@material-ui/core/IconButton'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

import MessageBadge from '../../components/message-badge'

describe('Message Badge Component', () => {
  let messageBadge

  beforeEach(() => {
    messageBadge = shallow(<MessageBadge />)
  })
  afterAll(() => jest.resetAllMocks())

  it('renders without crashing', () => {
    expect(toJson(messageBadge)).toMatchSnapshot()
  })

  it('should render a IconButton with length of 1', () => {
    expect(messageBadge.find(IconButton)).toHaveLength(1)
  })

  it('should find IconButton to be present', () => {
    expect(messageBadge.find(IconButton).exists()).toBe(true)
  })

  it('should render a IconButton componnent with a prop name `aria-label`', () => {
    expect(messageBadge.find(IconButton).prop('aria-label')).toBe('message')
  })

  it('should render a badge', () => {
    expect(messageBadge.find(Badge).exists()).toBeDefined()
  })

  it('should render a Box component', () => {
    expect(messageBadge.find(Box).exists()).toBe(true)
  })

  it('should render a Box component with a prop name `component`', () => {
    expect(messageBadge.find(Box).prop('component')).toBe('img')
  })

  it('should render a Box component with a prop name `src`', () => {
    expect(messageBadge.find(Box).prop('src')).toBeUndefined()
  })

  it('should render a Box component with a prop name `alt`', () => {
    expect(messageBadge.find(Box).prop('alt')).toBe('message')
  })
})
