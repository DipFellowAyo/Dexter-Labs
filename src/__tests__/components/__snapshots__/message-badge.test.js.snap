// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Badge Component renders without crashing 1`] = `
<WithStyles(ForwardRef(IconButton))
  aria-label="message"
>
  <WithStyles(WithStyles(ForwardRef(Badge)))
    badgeContent={1}
    color="secondary"
  >
    <Styled(MuiBox)
      alt="message"
      component="img"
    />
  </WithStyles(WithStyles(ForwardRef(Badge)))>
</WithStyles(ForwardRef(IconButton))>
`;
